src/aspire/storage/starfile.py

```diff
@@ -1,173 +1,135 @@
-import logging
 from collections import OrderedDict
 
 import pandas as pd
+from gemmi import cif
 
-logger = logging.getLogger(__name__)
 
-
-class StarFileBlock:
-    def __init__(self, loops, name="", properties=None):
-        # Note: StarFile data blocks may have have key=>value pairs that start with a '_'.
-        # We serve these up to the user using getattr.
-        # To avoid potential conflicts with our custom
-        # attributes here, we simply make them 'public' without a leading underscore.
-        self.loops = loops
-        self.name = name
-        self.properties = properties
-
-    def __repr__(self):
-        return f"StarFileBlock (name={self.name}) with {len(self.loops)} loops"
-
-    def __getattr__(self, name):
-        return self.properties[name]
-
-    def __len__(self):
-        return len(self.loops)
-
-    def __getitem__(self, item):
-        return self.loops[item]
-
-    def __eq__(self, other):
-        return (
-            self.name == other.name
-            and self.properties == other.properties
-            and all([all(l1 == l2) for l1, l2 in zip(self.loops, other.loops)])
-        )
+class StarFileError(Exception):
+    def __init__(self, message):
```

Comment:
If you are going do this I think you might just try the following, but I'd have to check... its been a while.

```
class StarFileError(Exception):
    pass
```

Reply:
Looks like it:
https://stackoverflow.com/questions/1319615/proper-way-to-declare-custom-exceptions-in-modern-python
(answer was updated march of this year)